storage/shard.go

```diff
@@ -732,7 +777,13 @@ func (s *dbShard) Write(
 	// If no entry and we are not writing new series asynchronously
 	if !writable && !opts.writeNewSeriesAsync {
 		// Avoid double lookup by enqueueing insert immediately
-		result, err := s.insertSeriesAsyncBatched(id, dbShardInsertAsyncOptions{})
+		result, err := s.insertSeriesAsyncBatched(id, tags, dbShardInsertAsyncOptions{
+			hasPendingIndexing: indexWriteFn != nil,
```

Comment:
the nil check seems a little weird, can we just throw an error if its nil?

Reply:
tl;dr - Unfortunately, no. 

Longer version - The `indexWriteFn` value can be nil in two situations: 
1) With indexing enabled, and a user calling `db.Write()` (not `db.WriteTagged()`). Need to indicate don't need to index in those situations. 
2) With indexing disabled, and we have code that calls `shard.WriteTagged()` directly. This will never happen in usual circumstances as `namespace.WriteTagged` checks if indexing is enabled before calling `shard.WriteTagged()`.

Reply:
Cool, can you just throw in a comment on when this can be nil then

Reply:
sure

Reply:
done